src/box/index_def.c

```diff
@@ -252,15 +252,11 @@ index_def_to_key_def(struct rlist *index_defs, int *size)
 	struct index_def *index_def;
 	rlist_foreach_entry(index_def, index_defs, link)
 		key_count++;
-	size_t bsize;
-	struct key_def **keys =
-		region_alloc_array(&fiber()->gc, typeof(keys[0]), key_count,
-				   &bsize);
-	if (keys == NULL) {
-		diag_set(OutOfMemory, bsize, "region_alloc_array", "keys");
-		return NULL;
-	}
 	*size = key_count;
+	if (key_count == 0)
+		return NULL;
+	struct key_def **keys =
+		xregion_alloc_array(&fiber()->gc, typeof(keys[0]), key_count);
```

Comment:
This function doesn't fail anymore. Please update the comment to the function declaration.

Reply:
Done.
```diff
  *
  * @param index_defs List head.
  * @param[out] size  Array size.
- * @retval not NULL  Array of pointers to key_def
- * @retval NULL      Memory error.
+ * @retval Array of pointers to key_def (NULL if size == 0).
  */
```